magpie/api/management/user/user_utils.py

```diff
@@ -872,6 +872,11 @@ def check_user_info(user_name=None, email=None, password=None, group_name=None,
         ax.verify_param(user_name, matches=True, param_name="user_name", param_compare=ax.PARAM_REGEX,
                         http_error=HTTPBadRequest,
                         msg_on_fail=s.Users_CheckInfo_UserNameValue_BadRequestResponseSchema.description)
+        supplemental_regex = get_constant("MAGPIE_SUPPLEMENTAL_USERNAME_REGEX", raise_missing=False)
+        if supplemental_regex:
+            ax.verify_param(user_name, matches=True, param_name="user_name", param_compare=supplemental_regex,
+                            http_error=HTTPBadRequest,
+                            msg_on_fail=s.Users_CheckInfo_UserNameValue_BadRequestResponseSchema.description)
```

Comment:
Is this `msg_on_fail` error message displayed also to the web UI or only in the logs?

Can I have a sample content of this `msg_on_fail` error message?  Is is understandable by someone not well versed with Magpie, like a new node admin.

Reply:
The value is defined here: https://github.com/mishaschwartz/Magpie/blob/a1bd0e15c88d353df51f5ca7988e7ffbc34d202f/magpie/api/schemas.py#L1961C1-L1961C57

and says: `"Invalid 'user_name' value specified."`

Reply:
Right, it does not say "why" and there is no way to add supplemental info that the reason is the new regex.  I hope the node admin will be able to guess it.

Reply:
> and there is no way to add supplemental info that the reason is the new regex

I can make or respond with a different message. How about: "Invalid 'user_name' specified. Does not match the supplemental user name regex."

Reply:
Sure that would be better if it does not take to much of your time.  Creating a new sub-class with a new hardcoded message?

Reply:
`msg_on_fail` is only the generic message value in the API body. 
For the "why", that would be reported in the API response body. It provides the `param_name`, `param_compare`, the input value and so on with more details about the specific check that failed. 

On the UI, it would simply be a red message next to the input text body because the contents are limited in the code. That could be improved.

Reply:
The message in the UI I mentioned is here: 
https://github.com/Ouranosinc/Magpie/blob/92ff2d2fa1e62b69a3a5edf22ef65d61238eaba9/magpie/ui/management/templates/add_user.mako#L39-L41
It uses the property obtained from this: 
https://github.com/Ouranosinc/Magpie/blob/92ff2d2fa1e62b69a3a5edf22ef65d61238eaba9/magpie/ui/management/views.py#L87-L88
Which default to this:
https://github.com/Ouranosinc/Magpie/blob/92ff2d2fa1e62b69a3a5edf22ef65d61238eaba9/magpie/ui/utils.py#L304-L308

That value could be overridden with more explicit details according to the contents parsed from the API response obtained here before returning the UI response:
https://github.com/Ouranosinc/Magpie/blob/92ff2d2fa1e62b69a3a5edf22ef65d61238eaba9/magpie/ui/management/views.py#L106-L108

Reply:
@fmigneault @tlvu with the current update to `msg_on_fail` the UI looks like this:

![image](https://github.com/Ouranosinc/Magpie/assets/4380924/2fa53a35-1b21-4981-99ac-c2c2a90a3292)

Is this sufficient or are there other changes you would suggest

Reply:
Still need rename of the variable/setting to `MAGPIE_USER_NAME_EXTRA_REGEX`. 
This is to align with other variable names, such as `MAGPIE_USER_NAME_MAX_LENGTH`, for similar checks.